src/libraries/System.Private.CoreLib/src/System/Number.NumberBuffer.cs

```diff
@@ -76,21 +76,12 @@ public void CheckConsistency()
                 uint integerDigitsPresent = Math.Min(positiveExponent, totalDigits);
                 uint fractionalDigitsPresent = totalDigits - integerDigitsPresent;
 
-                // Verify that the fractional part does not have any trailing zeros
-                int numberOfTrailingZeros = 0;
-                for (int i = DigitsCount - (int)fractionalDigitsPresent; i < DigitsCount; i++)
+                // For a number like 1.23000, verify that we don't store trailing zeros in Digits
+                // However, if the number of digits exceeds maxDigCount and rounding is required, we store the trailing zeros in the buffer.
```

Comment:
Does it require any more complex changes to just trim in this case as well?

I don't remember if this is going to have "cost" when dealing with the internal `BigInteger` (I think it does) and so it might be better to trim and have `HasNonZeroTail` correctly set if it doesn't require more complex changes.

Reply:
If it does require more changes, we can just log it as a "investigate later" issue

Reply:
I started investigating this a little bit last week. It looks like the number of fractional digits present is used in calculating a `fastExponent` which in turns determines whether we use the fast/slow path to convert a number to float/double. Also, `HasNonZeroTail` is only used in the slow path currently. For this change, I decided to just be conservative, but yea I'll log an "investigate later" issue. Would be something to work on in December (at the least).